Microsoft.Azure.CosmosRepository/src/FullItem.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) IEvangelist. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.CosmosRepository
+{
+    /// <summary>
+    /// A base helper class that implements IItemWithEtag
+    /// <remarks>
+    /// This base class will implement all supported interfaces giving the complete information available from this SDK.
+    /// </remarks>
+    /// </summary>
+    /// <example>
+    /// Here is an example subclass item, which adds several properties:
+    /// <code language="c#">
+    /// <![CDATA[
+    /// public class SubItem : FullItem
+    /// {
+    ///     public DateTimeOffset Date { get; set; }
+    ///     public string Name { get; set; }
+    ///     public IEnumerable<Child> Children { get; set; }
+    ///     public IEnumerable<string> Tags { get; set; }
+    /// }
+    ///
+    /// public class Child
+    /// {
+    ///     public string Name { get; set; }
+    ///     public DateTime BirthDate { get; set; }
+    /// }
+    /// ]]>
+    /// </code>
+    /// </example>
+    public abstract class FullItem : IItemWithEtag
```

Comment:
Again as well as implementing the interface it could also derive from `Item` save writing out the properties again